add Artery throughput and latency tests

patriknw opened the pull request:
This is just one first benchmark measuring max throughput. I will create others measuring latency.

drewhk commented on akka-remote-tests/src/multi-jvm/scala/akka/remote/artery/MaxThroughputSpec.scala:
```diff
@@ -0,0 +1,247 @@
+    def receive = {
+      case Start ⇒
+        c = 0
+        sender() ! Start
```
currently, you are better off caching the sender as it will point to a new `RemoteActorRef` instance every time, resulting in a lookup in the Association table for each send.

patriknw replied:
Yeah, i noticed that via OOME. That's why I changed the associate method.
In this test the replies are only for flow control but good to know for upcoming tests.

patriknw commented:
@drewhk I have fixed the use of FragmentAssembler, including the 32 fragment limit, see 8aedb615f9bb2bded58ede0b31d3024228774f88

patriknw committed:
improve Aeron source and sink

* use FragmentAssembler for large messages
* no need for aeron factory
* some logging/listeners of aeron events and exceptions

patriknw committed:
add Artery throughput and latency tests

* as multi-node tests
* adjust multi-node-testkit to use the hostname for Artery
* update HdrHistogram version

patriknw commented:
I have completed this. 4 tests added.

* AeronStreamMaxThroughputSpec throughput test for the low level streams
* AeronStreamLatencySpec latency test for the low level streams
* LatencySpec latency test for actor messages
* MaxThroughputSpec throughput test for actor messages

The bottom line results are prefixed with `===` for easy grep/search.

Note that by default the tests are only running for a very short time. For serious measurements you should set the config property `totalMessagesFactor`, e.g. `akka.test.MaxThroughputSpec.totalMessagesFactor = 20.0`

patriknw commented:
I will setup a separate jenkins nightly job that will run those tests with proper config.

ktoso commented:
A nice start for measuring, merge at will, we'll evolve those in time.
Btw, perhaps since they're not JMH benchmarks we should move them to a new module `akka-bench-multinode`?

patriknw commented:
I think they can be in akka-remote-tests, at least for now. note that they are quick to run until you configure them otherwise

patriknw commented:
I'll add my plot stuff here before merging.

ktoso commented:
> I'll add my plot stuff here before merging.

+1, keep scripts here and just invoke them from the build

patriknw committed:
add output for trend plots

* use Jenkins Plot plugin
* output csv values to log and then
  use post build step to turn those into csv files
  that the plot plugin understands

patriknw commented:
added the plot stuff, will merge after validation

patriknw commented:
I have setup a jenkins job that runs the performance tests twice per night. There I'm using the Jenkins Plot plugin to create trend graphs so that we can notice regressions and improvements. You find them at  https://jenkins.akka.io:8498/job/akka-artery-perf/plot/ (only includes two build runs so far).
